src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java

```diff
@@ -8,17 +8,16 @@
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.util.List;
-import java.util.Objects;
 
 
-public interface CSVToObjectService {
-    <T extends Object> List<T> parse(Reader csv);
+public interface CSVToObjectService<T> {
+    List<T> parse(Reader csv, Class<T> type);
 
     Logger getLogger();
 
-    default <T extends Object> List<T> parse(MultipartFile file) {
+    default List<T> parse(MultipartFile file, Class<T> type) {
         try {
-            return parse(new InputStreamReader(file.getInputStream()));
+            return parse(new InputStreamReader(file.getInputStream()), type);
```

Comment:
`new InputStreamReader(...)` opens a stream but never closes it.

Reply:
Fixed